Assets/Scripts3D/Weapons/Weapon.cs

```diff
@@ -4,19 +4,49 @@
 
 public class Weapon : MonoBehaviour
 {
-
     [SerializeField] Camera FPCamera;
     [SerializeField] float Range = 100f;
     [SerializeField] float damage = 20f;
     [SerializeField] ParticleSystem MuzzleFlash;
+    [SerializeField] Transform EquippedWeapon;
+
+    [SerializeField] PlayerController PlayerController;
+
+    private float movementCounter;
+    private float idleCounter;
+
+    private Vector3 _weaponOrigin;
+    private Vector3 _targetWeaponBobPosition;
+
+    [Header("Mouse Look Sway Settings")]
+    [SerializeField] private float _smooth;
```

Comment:
I've refactored this in branch [feature/rls-7-shotgun-behaviour](https://github.com/JamesSchiele/DungeonGenerator/tree/feature/rls-7-shotgun-behaviour) - base stuff is in Weapon.cs and I've created a subclass Shotgun.cs - please rebase off this branch and add any shotgun specific stuff to the latter script